From 2e21e2cd74085431c44389209284d9918d81f01c Mon Sep 17 00:00:00 2001
From: Tim Pope <code@tpope.net>
Date: Thu, 11 Sep 2008 17:43:22 -0400
Subject: [PATCH] Robust passing of autospec arguments to autotest

---
 bin/autospec |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/bin/autospec b/bin/autospec
index bcba414..dc47846 100755
--- a/bin/autospec
+++ b/bin/autospec
@@ -1,3 +1,3 @@
 #!/usr/bin/env ruby
 ENV['RSPEC'] = 'true'
-system (["autotest"] << ARGV.dup).flatten.join(" ")
\ No newline at end of file
+system ("autotest", *ARGV)
-- 
1.5.6.5


From 697f2dc926f17a34138712f3e47fe79f3d6367b5 Mon Sep 17 00:00:00 2001
From: Tim Pope <code@tpope.net>
Date: Thu, 11 Sep 2008 17:26:35 -0400
Subject: [PATCH] Only use color codes on tty; override for autospec

---
 bin/autospec                                     |    1 +
 lib/spec/runner/formatter/base_text_formatter.rb |    9 ++-------
 2 files changed, 3 insertions(+), 7 deletions(-)

diff --git a/bin/autospec b/bin/autospec
index dc47846..ee8d6bf 100755
--- a/bin/autospec
+++ b/bin/autospec
@@ -1,3 +1,4 @@
 #!/usr/bin/env ruby
 ENV['RSPEC'] = 'true'
+ENV['AUTOTEST'] = 'true'
 system ("autotest", *ARGV)
diff --git a/lib/spec/runner/formatter/base_text_formatter.rb b/lib/spec/runner/formatter/base_text_formatter.rb
index bad023d..5906d3c 100644
--- a/lib/spec/runner/formatter/base_text_formatter.rb
+++ b/lib/spec/runner/formatter/base_text_formatter.rb
@@ -15,11 +15,6 @@ module Spec
           super
           if where.is_a?(String)
             @output = File.open(where, 'w')
-          elsif where == STDOUT
-            @output = Kernel
-            def @output.flush
-              STDOUT.flush
-            end
           else
             @output = where
           end
@@ -81,7 +76,7 @@ module Spec
         end
         
         def close
-          if IO === @output
+          if IO === @output && @output != STDOUT
             @output.close 
           end
         end
@@ -112,7 +107,7 @@ module Spec
 
         def output_to_tty?
           begin
-            @output == Kernel || @output.tty?
+            @output.tty? || ENV.has_key?("AUTOTEST")
           rescue NoMethodError
             false
           end
-- 
1.5.6.5

