Index: rspec/spec/spec/interop/test/unit/passing_test_case.rb
===================================================================
--- rspec/spec/spec/interop/test/unit/passing_test_case.rb	(revision 0)
+++ rspec/spec/spec/interop/test/unit/passing_test_case.rb	(revision 0)
@@ -0,0 +1,5 @@
+require 'test/unit'
+
+class PassingTestCase < Test::UnitTestCase
+  def test_passing; end
+end
Index: rspec/spec/spec/interop/test/unit/testcase_spec.rb
===================================================================
--- rspec/spec/spec/interop/test/unit/testcase_spec.rb	(revision 3146)
+++ rspec/spec/spec/interop/test/unit/testcase_spec.rb	(working copy)
@@ -7,4 +7,10 @@
     output = run_script("#{dir}/testcase_spec_with_test_unit.rb")
     output.should include("3 examples, 0 failures")
   end
-end
\ No newline at end of file
+
+  it "should have the TestCase name in the output" do
+    dir = File.dirname(__FILE__)
+    output = run_script("#{dir}/passing_test_case.rb")
+    output.should include("PassingTestCase")
+  end
+end
