Index: rspec_on_rails/lib/spec/rails/example/view_example_group.rb
===================================================================
--- rspec_on_rails/lib/spec/rails/example/view_example_group.rb	(revision 3229)
+++ rspec_on_rails/lib/spec/rails/example/view_example_group.rb	(working copy)
@@ -28,6 +28,10 @@
           ensure_that_flash_and_session_work_properly
         end
 
+        after(:each) do
+          ensure_that_base_view_path_is_not_set_across_example_groups
+        end
+
         def initialize(defined_description, &implementation) #:nodoc:
           super
           @controller_class_name = "Spec::Rails::Example::ViewExampleGroupController"
@@ -40,11 +44,6 @@
           @controller.class.send :public, :flash
         end
 
-        def teardown #:nodoc:
-          super
-          ensure_that_base_view_path_is_not_set_across_example_groups
-        end
-
         def ensure_that_base_view_path_is_not_set_across_example_groups #:nodoc:
           ActionView::Base.base_view_path = nil
         end
Index: rspec_on_rails/spec/rails/example/view_spec_spec.rb
===================================================================
--- rspec_on_rails/spec/rails/example/view_spec_spec.rb	(revision 3229)
+++ rspec_on_rails/spec/rails/example/view_spec_spec.rb	(working copy)
@@ -243,7 +243,21 @@
           group.description.to_s.should == "foo"
           $nested_group.description.to_s.should == "foo bar"
         end
+
+        it "should clear ActionView::Base.base_view_path on teardown" do
+          ViewExampleGroup.class_eval do
+            alias_method(:ensure_that_base_view_path_is_not_set_across_example_groups_orig,
+              :ensure_that_base_view_path_is_not_set_across_example_groups)
+            define_method(:ensure_that_base_view_path_is_not_set_across_example_groups){
+              $base_view_path_cleared = true
+              ensure_that_base_view_path_is_not_set_across_example_groups_orig
+            }
+          end
+          describe("base_view_path_cleared flag", :type => :view) do
+            it { $base_view_path_cleared.should be_true }
+          end
+        end
       end
     end
   end
-end
\ No newline at end of file
+end
